src/EnergyPlus/CTElectricGenerator.cc

```diff
@@ -817,6 +762,67 @@ namespace CTElectricGenerator {
         }
     }
 
+    void CTGeneratorData::oneTimeInit(EnergyPlusData &state)
+    {
+        auto constexpr RoutineName("InitICEngineGenerators");
+        bool errFlag;
+
+        if (this->MyPlantScanFlag) { // this flag to be removed
```

Comment:
There are 2 logical expressions and 1 complex logical that will be executed each time this function is called. Would there be any performance improvement in wrapping these with a single logical expression or instead adding this at line 708? initOneTime is true until all 3 of these if blocks are executed.

if (this->initOneTime) this->oneTimeInit(state); // Do one-time inits